src/index.ts

```diff
@@ -7,6 +7,14 @@ import getRequestOptions from './get_request_options';
 
 export type ShouldRetry = (responseOrError: Response | Error) => boolean;
 
+export interface OnRequestEndOptions {
+  responseOrError: Response | Error;
+}
+
+export interface OnRequestOptions {
+  url: string;
+  retryCount: number;
+}
```

Comment:
nitpick: these names would be a little less misleading if you applied the intersection here instead of in the function below – do you know what I mean?
```suggestion
export interface OnRequestOptions extends JsonFetchOptions {
  url: string;
  retryCount: number;
}
```

Reply:
Yep